pkg/controller/release/release_controller.go

```diff
@@ -313,19 +313,12 @@ func (c *Controller) syncOneReleaseHandler(key string) error {
 		c.recorder,
 	)
 
-	// This is a 2-round handler: the 1st round schedules the release on a
-	// set of clusters, and the 2nd round creates associated objects and
-	// finalizes release scheduling process. This approach comes for a
-	// reason: we are eliminating a replication lag problem. While running
-	// the system in production, we observed some cases where a few
-	// sequentual updates on the same object experienced apiserver
-	// rejections due to the passed object outdate state. This happened due
-	// to the synchronisation/replication lag in etcd. This approach helps
-	// to eliminate this as a notification will be delivered once all
-	// parties are in sync.
 	if !releaseHasClusters(rel) {
-		_, err = scheduler.ChooseClusters(rel, false)
-	} else {
+		shouldForce := false
+		_, err = scheduler.ChooseClusters(rel, shouldForce)
```

Comment:
If you don't use the returned release here, the subsequent call to ScheduleRelease will always fail due to a conflict, as the resource versions of what you have in memory and what's in the apiserver will be different. That'd be equivalent to the previous version of the code.

Reply:
Fixed, thanks!